net/discovery.go

```diff
@@ -154,27 +156,46 @@ func (d *discovery) advertise() time.Duration {
 	return defaultAdvertiseTTL
 }
 
-func (d *discovery) discover(provides types.ProvidesCoin,
-	searchTime time.Duration) ([]peer.AddrInfo, error) {
-	log.Debugf("attempting to find DHT peers that provide [%s] for %vs...",
-		provides,
-		searchTime.Seconds(),
-	)
+func (d *discovery) discoverLoop() {
+	const discoverLoopDuration = time.Minute
+	timer := time.NewTicker(discoverLoopDuration)
+
+	for {
+		select {
+		case <-d.ctx.Done():
+			timer.Stop()
+			return
+		case <-timer.C:
+			if len(d.h.Network().Peers()) >= defaultMinPeers {
+				continue
+			}
+
+			// if our peer count is low, try to find some peers
+			timer := time.NewTimer(discoverLoopDuration)
+
+			_, err := d.findPeers("", timer.C)
+			if err != nil {
+				log.Errorf("failed to find peers: %s", err)
+			}
+
+			timer.Stop()
+		}
+	}
+}
 
-	peerCh, err := d.rd.FindPeers(d.ctx, string(provides))
+func (d *discovery) findPeers(provides string, done <-chan time.Time) ([]peer.AddrInfo, error) {
+	peerCh, err := d.rd.FindPeers(d.ctx, provides)
 	if err != nil {
 		return nil, err
 	}
 
-	timer := time.NewTicker(searchTime)
 	peers := []peer.AddrInfo{}
```

Comment:
If we are retesting this code right now, it would be a good time to update this declaration to `var peers []peer.AddrInfo`. Hopefully, we don't have any code that is treating nil and empty slices differently.

Reply:
updated